enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,306 @@
+Using a CR enable users to override the default backup configuration.
+
+By design, the _**No-Config**_ backup approach works _orthogonal_ to the [Cron](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups) based method, and not a replacement.
+However, this method is designed to work on all variants of Openshift (e.g. Single-node and Bare-metal), as well as having backups on all control-plane nodes, for recovering from disaster scenarios.
+
+### API Extensions
+
+This proposal uses the same [API](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions) from the Cron based method.
+No additional API changes are required.
+
+### Topology Considerations
+TBD
+#### Hypershift / Hosted Control Planes
+This proposal does not affect HyperShift.
+#### Standalone Clusters
+TBD
+#### Single-node Deployments or MicroShift
+TBD
+### Workflow Description
+TBD
+
+## Design Details
+
+To back up the Etcd datastore and cluster resources of an Openshift cluster for the purpose of restoration, a persistent storage must be utilised.
+Several main point have been considered during designing the implementation of this proposal, however the storage solution to be used is the most influential, particularly to be used by all Openshift variants.  
+
+### Storage
+
+Ideally, the storage solution to use should have the following characteristics :-
+
+- Backups should be taken on each master node to avoid overwhelming a specific etcd member more than others.
+- Spreading the backups among all etcd cluster members provide guarantees for disaster recovery in case of losing two members at the same time.
+- One fits all storage solution, to be used across all Openshift variants including Bare metal and Single Node Openshift.
+
+According to these requirement, only `hostPath` and `localVolume` storage could be utilized. 
+However, for `localVolume` storage, if the master node where the `PV` is mounted became unhealthy/unavailable/unreachable, then the backups are no longer accessible, also taking new backups is no longer possible.
+Hence, `hostPath` has been used as it is the only viable option that could be utilized on all Openshift variants. 
+
+### Workload
+
+In order to take backups for Etcd data store and cluster resources, a workload must run on the cluster while having access to Etcd and cluster resources.
+Since Etcd data store of an Openshift cluster runs on `Pod` similar to other workload, several constructs such as `Deployment`, `DaemonSet`, `StatefulSet` and `SideCar` could be used.
+
+This section discusses two approaches (i.e. `SideCar` and `DaemonSet`), which are implemented and tested, explaining the Pros and Cons of each method.  
+Before deciding upon these two methods, several other approaches have been investigated. See [Alternatives](#Alternatives).
+
+#### SideCar container within all Etcd-static-pod
```

Comment:
I would appreciate your final opinion, which approach should be used and merged ? 

I personally vote for the `DaemonSet` approach, for the following 

- It mitigates  the issues of the `StaticPod` approach, such as rolling out a new revision upon enabling / disabling the `etcd-backup-server`

Reply:
I like daemonset approach. With 2 node OpenShift we can make sure advisor node is excluded - this approach is more flexible than sidecar

Reply:
I also think the same, I was astonished by how stable it is. 

I think it worth testing this during Cluster-Disruption 

So we make noise on the control-plane nodes, and see how this affects the Backup pods, wdyt ? 

I think if the Etcd cluster is under load, we should shut down the backup, because this will overload the cluster, which is already suffering

Reply:
Move this to alternatives, let's go with daemonset